src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateWithClones = [];
+  let stateClone = { ...state };
+
+  for (const types of actions) {
+    switch (types.type) {
+      case 'addProperties':
+        stateClone = Object.assign(stateClone, types.extraData);
+
+        break;
+
+      case 'removeProperties':
+        for (const removeKey of types.keysToRemove) {
+          delete stateClone[`${removeKey}`];
+        }
+        break;
+
+      case 'clear':
+        stateClone = {};
+        break;
+      default:
```

Comment:
add blank lines between cases
```suggestion
        break;

      default:
```